middleware/validationSchemas.js

```diff
@@ -34,38 +36,149 @@ export const createUserSchema = Joi.object({
     "string.min": "Password should have a minimum length of {#limit}.",
     "string.max": "Password should have a maximum length of {#limit}.",
-    "any.required": "Password is required.",
   }),
-  money: Joi.number().required().messages({
-    "number.base": "Money must be a number.",
-    "number.unsafe": "Money is outside of usable range of numbers.",
-    "any.required": "Money is required.",
+});
+
+
+
+const itemRegex = /^[A-Za-z0-9_\(\)]+$/;
+const itemTypeRegex = /^[A-Z]+$/;
+
+export const createItemSchema = Joi.object({
+  name: Joi.string().min(1).max(25).regex(itemRegex).required().messages({
+    "string.base": "Name should be a string.",
+    "string.pattern.base": "Name can only consist of Alphanumeric and '!@#$%^&*()_-?' characters.",
+    "string.empty": "Name cannot be empty.",
+    "string.min": "Name should have a minimum length of {#limit}.",
+    "string.max": "Name should have a maximum length of {#limit}.",
+    "any.required": "Name is required.",
+  }),
+  item_type: Joi.string().regex(itemTypeRegex).required().messages({
+    "string.base": "Item type should be a string.",
+    "string.pattern.base": "Item type can only consist of captial letters.",
+    "string.empty": "Item type cannot be empty.",
+    "any.required": "Item type is required.",
+  }),
+  cost: Joi.number().integer().min(-intLimit).max(intLimit).options({ convert: false }).required().messages({
+    "number.base": "Cost must be a number.",
+    "number.integer": "Cost must be an integer.",
+    "number.max": "Cost is larger than usable Integer 32 range.",
+    "number.min": "Cost is smaller than usable Integer 32 range.",
+    "number.unsafe": "Cost is outside of usable range of numbers.",
+    "any.required": "Cost is required.",
   }),
 });
 
-export const updateUserSchema = Joi.object({
-  name: Joi.string().min(1).max(25).regex(nameRegex).required().messages({
+export const updateItemSchema = Joi.object({
+  name: Joi.string().min(1).max(25).regex(itemRegex).messages({
     "string.base": "Name should be a string.",
     "string.pattern.base": "Name can only consist of Alphanumeric and '!@#$%^&*()_-?' characters.",
     "string.empty": "Name cannot be empty.",
     "string.min": "Name should have a minimum length of {#limit}.",
     "string.max": "Name should have a maximum length of {#limit}.",
     "any.required": "Name is required.",
   }),
-  password: Joi.string().min(8).max(128).messages({
-    "string.base": "Password should be a string.",
-    "string.empty": "Password cannot be empty.",
-    "string.min": "Password should have a minimum length of {#limit}.",
-    "string.max": "Password should have a maximum length of {#limit}.",
+  item_type: Joi.string().regex(itemTypeRegex).messages({
+    "string.base": "Item type should be a string.",
+    "string.pattern.base": "Item type can only consist of captial letters.",
+    "string.empty": "Item type cannot be empty.",
+    "any.required": "Item type is required.",
```

Comment:
Thanks for pointing this out! This is actually the reverse and a problem with me copying and pasting without checking properly. The schemas that don't have `.required()` should have them and will add them instead of removing the messages!

Reply:
But these are update schemas. Adding on `.required()` would make it so you have to include every field in the JSON request when you update an object, which is not necessary